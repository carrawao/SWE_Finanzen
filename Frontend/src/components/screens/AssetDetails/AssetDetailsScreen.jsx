import React from 'react';
import {useParams} from 'react-router-dom';
import ScreensTemplate from '../../ScreensTemplate';
import StockchartCard from './StockchartCard';
import PropTypes from 'prop-types';

const AssetDetailsScreen = props => {
  let {asset, assetType} = useParams();  
  //data that we saved for this asset --> undefined if asset is not in the portfolio
  const savedAssetData = asset === undefined ? undefined : props.portfolioData[props.activePortfolio][assetType === 'Crypto' ? 'crypto' : 'shares'].find(element => element.symbol === asset);
  console.log(savedAssetData);
  const renderBody = () => (
    <React.Fragment>
      <h1>Details of asset: {assetType}|{asset} </h1>
      <StockchartCard assetType={assetType} symbol={asset}/>
    </React.Fragment>
  );

  return (
    <React.Fragment>
      <ScreensTemplate
        bodyComponent={renderBody}
        selectedNavLinkIndex={-1}
        assetsListArray={props.assetsListArray}
        searchResult={props.searchResult}
        setSearchResult={props.setSearchResult}
      />
    </React.Fragment>
  );
}

AssetDetailsScreen.propTypes = {
  searchResult: PropTypes.array,
  setSearchResult: PropTypes.func,
  portfolioData: PropTypes.object,
  activePortfolio: PropTypes.string,
  watchListsArray: PropTypes.array,
  assetsListArray: PropTypes.array,
};

export default AssetDetailsScreen;
