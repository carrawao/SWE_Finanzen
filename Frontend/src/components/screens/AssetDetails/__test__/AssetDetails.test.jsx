import React from 'react';
import AssetDetailsScreen from '../AssetDetailsScreen';
import toJson from 'enzyme-to-json';
import Enzyme, {shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() })

it('AssetDetailsScreen renders without crashing', () => {
  const portfolioData = {
    "Portfolio": {
      "name": "Portfolio",
      "value": 0,
      "gains": 0,
      "realisedGains": 0,
      "performance": 0,
      "shares": [],
      "crypto": [],
      "cash": [],
      "activities": [],
      "updated": "1970-01-01"
    },
  };

  const setSearchResult = jest.fn();

  const wrapper = shallow(
    <AssetDetailsScreen
      searchResult={[]}
      setSearchResult={setSearchResult}
      watchListsArray={[]}
      assetsListArray={[[]]}
      portfolioData={portfolioData}
      activePortfolio={'Portfolio'}
    />);

  expect(toJson(wrapper)).toMatchSnapshot();
})