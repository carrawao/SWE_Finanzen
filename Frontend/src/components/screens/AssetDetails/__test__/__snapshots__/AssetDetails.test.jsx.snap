// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetDetailsScreen renders without crashing 1`] = `
<Fragment>
  <ScreensTemplate
    assetsListArray={
      Array [
        Array [],
      ]
    }
    bodyComponent={[Function]}
    searchResult={Array []}
    selectedNavLinkIndex={-1}
    setSearchResult={[MockFunction]}
  />
</Fragment>
`;
